src/systems/filecoin_mining/storage_mining/sector_state_enum.go

```diff
@@ -0,0 +1,62 @@
+package storage_mining
+
+// SectorState is an enum of
+// SectorCommitted, SectorRecovering, SectorActive, SectorFailing, SectorCleared
+// FaultCount is only relevant for SectorRecovering and SectorFailing.
+// FaultCount is relevant in accounting for the number
+// of consecutive proving periods that a sector is Failing.
+// Sectors that are in Failing for more than CONSECUTIVE_FAULT_COUNT_LIMIT
+// in a row will result in Sectors getting cleared and miners penalized.
+// The enum is written in this awkward way because of golang limitation.
+
+const MAX_CONSECUTIVE_FAULT_COUNT uint8 = 3
+
+const (
+	SectorCommittedStateNo  uint8 = 0
+	SectorRecoveringStateNo uint8 = 1
+	SectorActiveStateNo     uint8 = 2
+	SectorFailingStateNo    uint8 = 3
+	// SectorClearedStateNo    uint8 = 4 // deleted from sm
```

Comment:
Still have it as a state, and make this state 0 (the null state). 

```go
const (
	SectorClearedStateNo    uint8 = 0
	SectorCommittedStateNo  uint8 = 1
	SectorActiveStateNo     uint8 = 2
	SectorRecoveringStateNo uint8 = 3
	SectorFailingStateNo    uint8 = 4
)
```

(also reordered a couple)